internal/cli/run.go

```diff
@@ -635,6 +645,32 @@ func (s Service) attemptRetries(
 				s.Log.Warn(err)
 			}
 		}
+
+	FLATTENED_TEST_RESULTS:
+		for _, originalTest := range flattenedTestResults.Tests {
```

Comment:
I wonder if this would be more clear written as `originalTestResults` - I recognize they're the same thing at this point though (as an aside, maybe we should pull both `flattenedTestResults` and `flattenedNewlyExecutedTestResults` down after this new loop where we actually assign values to them and change them

Don't feel like you have to change anything, though - it just took me a sec without the context of the code higher up above this loop

Reply:
Hm, I think we need this to be `flattenedTestResults` though.

For the first retry, `originalTestResults` and `flattenedTestResults` are the same. But for subsequent retries, we want to compare results against the merged result set, not the original one.

For example, you might have the following scenario:

**Original**
Test 1: Failure
Test 2: Failure

**Retry 1**
Test 1: Success
Test 2: Failure

**Retry 2**
Test 2: Success

I.e. we do want to merge the original results and the first retry before comparing against the results of the second retry - otherwise Captain could identify "Test 1" to be missing from the second retry, even though it succeeded previously.